### PR TITLE
implement :det_exact and :inverse_exact for list/yale matrices

### PR DESCRIPTION
**Pull request by spacewander**
This pr is about issue https://github.com/SciRuby/nmatrix/issues/189, which succeeds https://github.com/SciRuby/nmatrix/pull/334

### CODE REVIEW

**Comment by mohawkjohn** [comment]
@cjfuller Can you think of any cases of sparse Yale 3x3 matrices that aren't covered by these tests?

Thanks, @spacewander!

**Comment by mohawkjohn** [comment]
@spacewander Can you include some specs with Yale 3x3 matrices with more zeros? I *think* your tests are providing full coverage, but I'm not one hundred percent certain.

**Comment by spacewander** [comment]
No problem.

**Comment by mohawkjohn** [comment]
@spacewander Any progress on this?

**Comment by spacewander** [comment]
Oh yes, I have added a new case to this pr:
```
c = NMatrix.new(3, [1,0,3,0,0,1,0,6,0], stype: :yale, dtype: :int64)
inversed = c.method(:__inverse_exact__).call(c.clone, 3, 3)
d = NMatrix.new(3, [1,-3,0,0,0,0,0,1,0], stype: :yale, dtype: :int64)
expect(inversed).to eq(d)
```
And the code can work properly.

Is it enough?

File: spec/00_nmatrix_spec.rb [hunk 1]
```diff
@@ -41,8 +41,68 @@
     expect { n[0] }.to raise_error(ArgumentError)
   end
 
-  it "calculates exact determinants on small square matrices" do
+  it "calculates exact determinants on small dense matrices" do
```
Comment (mohawkjohn) [inline spec/00_nmatrix_spec.rb]: square dense

**Comment by mohawkjohn** [comment]
This seems to be missing a spec. If someone wants to write a spec for it, it'd be ready for review, I think. For GSOC students, we'll also accept a spec showing this pull request *doesn't* work as your code submission to NMatrix.

**Comment by mohawkjohn** [comment]
Overcome by #580.

**Pull request closed without merging**